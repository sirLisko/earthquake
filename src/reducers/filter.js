const initState = {};

export const FILTER_SET = 'FILTER_SET';
export const FILTER_CLEAR = 'FILTER_CLEAR';

export const setFilter = data => ({
  type: FILTER_SET,
  payload: data,
});

export const clearFilter = data => ({
  type: FILTER_CLEAR,
});

export const getCurrentFilter = state => state.filter;
export const getCurrentFilterName = state => state.filter.name;

export default (state = initState, action) => {
  switch (action.type) {
    case FILTER_CLEAR:
      return {};
    case FILTER_SET:
      return { ...state, ...action.payload };
    default:
      return state;
  }
};
