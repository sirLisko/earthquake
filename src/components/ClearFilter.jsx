import React from 'react';
import { useDispatch } from 'react-redux';

import { clearFilter } from 'reducers/filter';

const ClearFilter = () => {
  const dispatch = useDispatch();
  return (
    <button onClick={() => dispatch(clearFilter())} type="submit">
      clear filter
    </button>
  );
};

export default ClearFilter;
