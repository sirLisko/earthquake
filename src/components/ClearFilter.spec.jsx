import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { mount } from 'enzyme';

import * as filter from 'reducers/filter';

import ClearFilter from './ClearFilter';

const mockStore = configureStore([]);

describe('ClearFilter Component', () => {
  let wrapper;
  const emptyStore = mockStore({ filter: {} });

  beforeEach(() => {
    jest.spyOn(filter, 'clearFilter');
    wrapper = mount(
      <Provider store={emptyStore}>
        <ClearFilter />
      </Provider>,
    );
  });

  it('should render properly', () => {
    expect(wrapper.find('ClearFilter')).toMatchSnapshot();
  });

  it('should dispatch the correct action', () => {
    wrapper.find('button').simulate('click');
    expect(filter.clearFilter).toHaveBeenCalled();
  });
});
