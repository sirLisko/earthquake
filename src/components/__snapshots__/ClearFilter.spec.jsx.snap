// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClearFilter Component should render properly 1`] = `
<ClearFilter>
  <button
    onClick={[Function]}
    type="submit"
  >
    clear filter
  </button>
</ClearFilter>
`;
